### PR TITLE
App details view multicluster

### PR DESCRIPTION
**Pull request by hhovsepy**
Subtask https://github.com/kiali/kiali/issues/5966

### CODE REVIEW

File: frontend/src/pages/AppDetails/AppDetailsPage.tsx [hunk 1]
```diff
@@ -62,7 +62,7 @@ class AppDetails extends React.Component<AppDetailsProps, AppDetailsState> {
   constructor(props: AppDetailsProps) {
     super(props);
     const urlParams = new URLSearchParams(this.props.location.search);
-    const cluster = urlParams.get('cluster') || HomeClusterName;
+    const cluster = urlParams.get('cluster') || undefined;
```
Comment (leandroberetta) [inline frontend/src/pages/AppDetails/AppDetailsPage.tsx:65]: Shouldn't be the home cluster by default if the cluter query parameter is missing?.

Comment (hhovsepy) [reply frontend/src/pages/AppDetails/AppDetailsPage.tsx:65]: Originally the HomeClusterName in frontend was empty '' string and supposed to be removed,
we have it now by default in backend, when no cluster param is send, the backend takes the default conf.KubernetesConfig.ClusterName

File: frontend/src/pages/AppDetails/AppDetailsPage.tsx [hunk 2]
```diff
@@ -150,7 +150,12 @@ class AppDetails extends React.Component<AppDetailsProps, AppDetailsState> {
   private staticTabs() {
     const overTab = (
       <Tab title="Overview" eventKey={0} key={'Overview'}>
-        <AppInfo app={this.state.app} duration={this.props.duration} health={this.state.health} />
+        <AppInfo
+          app={this.state.app}
+          duration={this.props.duration}
+          health={this.state.health}
+          cluster={this.state.cluster ? this.state.cluster : ''}
```
Comment (josunect) [inline frontend/src/pages/AppDetails/AppDetailsPage.tsx:157]: If cluster is part of app, probably the cluster parameter is not needed here?

File: frontend/src/pages/AppDetails/AppInfo.tsx [hunk 3]
```diff
@@ -51,7 +52,12 @@ class AppInfo extends React.Component<AppInfoProps, AppInfoState> {
     if (!this.props.app) {
       return;
     }
-    this.graphDataSource.fetchForVersionedApp(this.props.duration, this.props.app.namespace.name, this.props.app.name);
+    this.graphDataSource.fetchForVersionedApp(
+      this.props.duration,
+      this.props.app.namespace.name,
+      this.props.app.name,
+      this.props.cluster
```
Comment (josunect) [inline frontend/src/pages/AppDetails/AppInfo.tsx:59]: So it should change here: 

`this.props.app.cluster ? this.props.app.cluster : ''`

**Review by josunect — COMMENTED** [review]
Tested and works as expected. I've just added a couple of comments

File: frontend/src/pages/AppDetails/AppDetailsPage.tsx (hunk 2 above)
Comment (hhovsepy) [reply frontend/src/pages/AppDetails/AppDetailsPage.tsx:157]: could be optimized, it is done similar to workloads and services, then we need to sync them as well

**Review by leandroberetta — APPROVED** [review]
LGTM

**Review by josunect — APPROVED** [review]
Thanks, @hhovsepy.
I haven't tested again but the changes looks good.

**Pull request merged**